packages/api/core/src/util/forge-config.ts

```diff
@@ -123,8 +124,12 @@ export default async (dir: string): Promise<ResolvedForgeConfig> => {
 
   if (await forgeConfigIsValidFilePath(dir, forgeConfig)) {
     try {
+      // The loaded "config" could potentially be a static forge config, ESM module or async function
       // eslint-disable-next-line @typescript-eslint/no-var-requires
-      const loaded = require(path.resolve(dir, forgeConfig as string)) as MaybeESM<ForgeConfig>;
+      let loaded = require(path.resolve(dir, forgeConfig as string)) as MaybeESM<ForgeConfig> | AsyncFn;
+      if (typeof loaded === 'function') {
+        loaded = await loaded();
```

Comment:
I think this has to be done _after_ we pull out `default` below.  Otherwise ESM config files that do `export default async function() { return { } }` won't work (but should)

Reply:
Oof, you're right, thanks! Will fix that